modules/dnn/src/layers/softmax_layer.cpp

```diff
@@ -75,9 +76,14 @@ class SoftMaxLayerImpl CV_FINAL : public SoftmaxLayer
 
     SoftMaxLayerImpl(const LayerParams& params)
     {
-        axisRaw = params.get<int>("axis", 1);
+        if(params.has("opset") && params.get<int>("opset") > 11){
+            axisRaw = params.get<int>("axis", -1);
+        }else{
+            axisRaw = params.get<int>("axis", 1);
+        }
         logSoftMax = params.get<bool>("log_softmax", false);
         setParamsFrom(params);
+        opt.init();
```

Comment:
1. Is `axis` normalized somewhere? I could not find it.
2. I propose to put `opt.init()` in `finalize` to keep alignment with that in gemm.

Reply:
1. the `axis` normalized in the `softmax` kernel.
2. 👌